README.md

```diff
@@ -141,34 +141,44 @@ List<Project> allProjects = projectPager.all();
 
 ---
 ## Java 8 Stream Support
-As of GitLab4J-API 4.9.2, you can also stream list based items in a Java 8 Stream using a Pager instance as follows:
+As of GitLab4J-API 4.9.2, all GitLabJ-API methods that return a List result also similarlly named method returns a Java 8 Stream.  The Stream returning methods use the following naming convention:  ```getXxxxxStream()```.  For example:
+
 ```java
-// Get a Pager instance to get a Stream<Project> instance.
-Pager<Project> projectPager = gitlabApi.getProjectsApi().getProjects(10);
+// List method
+List<Project> projects = gitlabApi.getProjectApi().getProjects();
 
-// Stream the Projects printing out the project name.
-projectPager.stream().map(Project::getName).forEach(name -> System.out.println(name));
-```
-The following API classes also include ```getXxxxxStream()``` methods which return a Java 8 Stream:
-```
-GroupApi
-IssuesApi
-NotesApi
-ProjectApi
-RepositoryApi
-TagsApi
-UserApi
+// Stream method
+Stream projectsStream<Project> = gitlabApi.getProjectApi().getProjectsStream();
 ```
+
 Example usage:
+
 ```java
 // Stream the visible Projects printing out the project name.
-gitlabApi.getProjectsApi().getProjectsStream().map(Project::getName).forEach(name -> System.out.println(name));
+Stream projectsStream<Project> = gitlabApi.getProjectApi().getProjectsStream();
+projectsStream.map(Project::getName).forEach(name -> System.out.println(name));
 
 // Operate on the stream in parallel, this example sorts User instances by username
 Stream<User> stream = new UserApi(gitLabApi).getUsersStream();
-List<User> sortedUsers = stream.parallel().sorted(comparing(User::getUsername)).collect(toList());
+List<User> users = stream.parallel().sorted(comparing(User::getUsername)).collect(toList());
+```
+
+**IMPORTANT**  
+The built-in methods that return a Stream do so using ___eager evaluation___, meaning all items are pre-fetched from the GitLab server and a Stream is returned which will stream those instances.  This allows for paralell processing of the Stream.
+
+To stream using ___lazy evaluation___, utilize the GitLab4J-API methods that return a ```Pager``` instance, and then call the ```lazyStream()``` method on the ```Pager``` instance to create a lazy evaluation Stream.
```

Comment:
I will add information about how it works: Lazy stream is fetching data page by page and can stop earlier when a limit is specified on stream. Fetching always starts from the beginning, so it cannot skip elements. This stream doesn't support parallel operations.

Reply:
I added some info regarding lazy streams with the latest commits.